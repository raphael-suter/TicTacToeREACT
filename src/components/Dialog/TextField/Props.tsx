export default interface Props {
  label: string;
  placeholder: string;
  value: string;
  isValid: boolean;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}
