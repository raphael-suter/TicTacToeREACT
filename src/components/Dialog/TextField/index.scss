@import "../../../styles/colors.scss";

.dialog-textfield {
  display: block;
  margin: 0.6rem;
  font-weight: bold;
  color: $darkColor;

  input {
    width: 100%;
    display: block;
    margin-top: 0.5rem;
    padding: 0.7rem;
    font-size: 12pt;
    border: 1px solid $mainColor;
    border-radius: 0;
    box-sizing: border-box;

    &.empty {
      border-color: red;
    }
  }
}
