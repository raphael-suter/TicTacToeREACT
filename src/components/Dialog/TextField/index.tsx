import React from 'react';
import './index.scss';
import Props from './Props';

export default ({ label, placeholder, value, isValid, onChange }: Props) => (
  <label className='textfield'>
    {label}
    <input className={`textfield__input ${isValid ? '' : 'textfield__input--invalid'}`} type='text' placeholder={placeholder} value={value} onChange={onChange} />
  </label>
);
