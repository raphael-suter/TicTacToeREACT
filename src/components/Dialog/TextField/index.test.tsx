import { shallow } from 'enzyme';
import React from 'react';
import TextField from '.';

const setup = (label: string, placeholder: string, value: string, isValid: boolean, onChange: (event: React.ChangeEvent<HTMLInputElement>) => void) => shallow(<TextField label={label} placeholder={placeholder} value={value} onChange={onChange} isValid={isValid} />);

describe('TextField', () => {
  test('Renders without crashing.', () => {
    const wrapper = setup(null, null, null, null, null);
    expect(wrapper).toHaveLength(1);
  });

  test('Displays label.', () => {
    const expectedText = 'qwertz';
    const wrapper = setup(expectedText, null, null, null, null);
    const receivedText = wrapper.text();

    expect(receivedText).toBe(expectedText);
  });

  test('Displays placeholder.', () => {
    const expectedText = 'qwertz';
    const wrapper = setup(null, expectedText, null, null, null);
    const receivedText = wrapper.childAt(0).props().placeholder;

    expect(receivedText).toBe(expectedText);
  });

  test('Displays value.', () => {
    const expectedText = 'qwertz';
    const wrapper = setup(null, null, expectedText, null, null);
    const receivedText = wrapper.childAt(0).props().value;

    expect(receivedText).toBe(expectedText);
  });

  test('Displays if valid.', () => {
    const wrapper = setup(null, null, null, true, null);
    const isValid = wrapper.childAt(0).props().isValid;

    expect(isValid).toBeTruthy();
  });

  test('Displays if invalid.', () => {
    const wrapper = setup(null, null, null, false, null);
    const isValid = wrapper.childAt(0).props().isValid;

    expect(isValid).toBeFalsy();
  });

  test('Fires change event.', () => {
    const onChange = jasmine.createSpy();
    const wrapper = setup(null, null, null, null, onChange);

    wrapper.childAt(0).simulate('change');

    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
