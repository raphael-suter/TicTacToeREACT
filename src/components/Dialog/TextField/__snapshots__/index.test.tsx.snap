// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Renders correctly. 1`] = `
<styled.label>
  text
  <styled.input
    isValid={true}
    onChange={null}
    placeholder="text"
    type="text"
    value="text"
  />
</styled.label>
`;
